services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/AttachmentValidator.kt

```diff
@@ -16,16 +16,21 @@ private val supportedFiletypes =
     )
 
 object AttachmentValidator {
-    fun validate(input: MultipartFile): MultipartFile {
-        val fileName = FileNameValidator.validateFileName(input.originalFilename)
+    fun validScanInput(attachment: MultipartFile): FileScanInput {
+        val fileName = FileNameValidator.validFileName(attachment.originalFilename)
+        val contentType = attachment.contentType.orEmpty()
```

Comment:
I'm having a hard time following what happens with null values in `attachment.contentType`. Here we change it to `""` if it's null. Then we give it to `contentTypeMatchesExtension` which can explicitly handle null values just fine. Then we log with the value just missing: `File liite.pdf extension does not match content type `.

For FileScanInput and further processing the value is taken from `attachment.contentType` again.

Reply:
I cleaned the contentype handling. Currently we don't actually need it anywhere apart from the file name validation.